app/models/ability.rb

```diff
@@ -0,0 +1,15 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    if user.is? :admin
+      can :manage, :all
+    else
+      can :read, :all
+      can :create, Post
+      can :create, Comment
+      can :delete, Post, author_id: user.id
+      can :delete, Comment, user_id: user.id
+    end
+  end
```

Comment:
- [ ]  Great job. Kindly add `user ||= User.new ` for the guest user who is not logged in yet. Something like this. :point_down: 

```
  def initialize(user)

         user ||= User.new # guest user (not logged in)

         if user.admin? 
         can :manage, :all
         else
         can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user
         can :read, :all
         end
     end

```

Reply:
Hi @Qoosim thank you for the review.
Changes done ✅ :
- added Cancancan to load and authorize resources on posts and comments controllers.
- added guest user for those who are not logged in yet.

Reply:
Addressed